exercises/randomgenerator.js

```diff
@@ -1,22 +1,35 @@
 
-
 class RandomGenerator{
 	constructor(min, max){ 
 		this.min = min;
+		if (min = undefined) {
```

Comment:
Another common thing in object creation, (and programming at large) is having default values available for functions to use if no other values are available. In this example of a random number generator, we want the generator to be able to generate a number between the min and max number we supply to it, or to generate a number between 1 and 10 if we do not supply the number. The approach used here is fine, but we can achieve the same functionality in a different way using default values. For example, instead of using a conditional check to see if the “min” and “max” variables are “undefined, you could use a “default or || statement” to check if there is a value to min, and if not, to assign a default value: 

```javascript
    constructor(min, max){
this.min = min || 1
this.max = max || 10;
this.random = null;
}
```
This example will assign to “this.min” what was passed in if a value was passed in, or a 1 if no value was passed in. But, there is an issue with this solution. The issue is that if a “falsey” value is passed in, (like the number zero), the “or” || statement will always choose the default value, (in this case 1). This is a very important thing to consider when using “or” || statements, as any falsy value will cause them to use the default value. We can fix this nature in this example by creating a conditional to account for the “zero” being passed in. for example:

```javascript
if(min === 0) {
this.min = min
} else {
this.min = min || 0
this.max = max || 10
```
Now if zero is passed in, the the conditional will handle it.